### PR TITLE
Add Style guide

### PR DESCRIPTION
**Pull request by raklaptudirm**
Add go Style and Idiom guide for contributors.

### CODE REVIEW

**Comment by raklaptudirm** [comment]
@tjgurwara99 @siriak @yanglbme You guys can also commit any guidelines you may have.

**Review by yanglbme — DISMISSED** [review]

**Comment by raklaptudirm** [comment]
@yanglbme We are still working on this guide, so keep this as a draft.

**Comment by raklaptudirm** [comment]
@tjgurwara99 @siriak @yanglbme I am currently busy with some exams. In the meantime, you all can discuss and add changes here.

File: STYLE.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+Package names should be short and to the point. Keep in mind that this identifier will be used to
+refer to your package, so make it easy to remember. It should be only composed of lower case
+letters. Prefer `json` to `JSON` or `Json`. If your package name has two words, merge them
+together. Prefer `jsonencodeing` to `jsonEncoding` or `json_encodeing`.
+
+Add the `_test` suffix to your package name to implement black-box testing for your package.
+
+### Symbol Naming
+
+Go symbols should be named in the `camelCase` or `UpperCamelCase`, depending of whether the symbol
```
Comment (tjgurwara99) [inline STYLE.md:149]:
```suggestion
Go symbols should be named in the `camelCase` or `PascalCase`, depending of whether the symbol
```

File: STYLE.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+For exported symbols, use the package name to your advantage to write concise symbol names. For
+example, if you have a package `png`, instead of defining a `png.PNGFile`, define a `png.File`.
+It is much clearer and avoids repetition.
+
+### Function Naming
+
+#### Constructor Naming
+
+Constructors should use the naming format `New<Type>()`. If the package only exports a single
```
Comment (tjgurwara99) [inline STYLE.md:161]: This is somewhat relevant, `New<Something>` should return a pointer to be consistent with `new` in built function and `Make<Something>` should return an object to be consistent with `make` in built function.

File: STYLE.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+constructor, use the name `New()`. Some valid names are `reader.New()`, `metainfo.NewFile()`.
+
+#### Getter Naming
```
Comment (tjgurwara99) [inline STYLE.md:164]: Maybe add a note here to avoid getters and setters as much as you can

**Review by akshitagupta15june — DISMISSED** [review]

**Review by Cali1093 — DISMISSED** [review]

**Comment by siriak** [comment]
@raklaptudirm @tjgurwara99 lets get this resolved

**Comment by raklaptudirm** [comment]
Completing the Style Guide will be a time-consuming task. I currently do not have the time really to finish this project. However, we have made pretty good progress on it, and I think this can be merged for now. In the future, we can expand on it. However, feel free to make any more changes.

**Review by siriak — APPROVED** [review]
Sure, existing @tjgurwara99 's comments need to be resolved and this can be merged

File: STYLE.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+      - [Getter Naming](#getter-naming)
+      - [Setter Naming](#setter-naming) 
+  - [Interface Naming](#interface-naming)
+
+## Formatting
+
+All go code should be formatted with the official formatting tool `gofmt`. This requirement is
+implemented and verified by the repository workflows.
+
+It is also required for the code to pass the `go vet` checks.
```
[the remark this answers is not in the document]
Comment (siriak) [reply STYLE.md:29]: Neither do I. Let's just remove this sentence for the time being.

**Review by siriak — APPROVED** [review]

**Review by tjgurwara99 — APPROVED** [review]

**Review by cclauss — APPROVED** [review]

**Review by tjgurwara99 — APPROVED** [review]

**Pull request merged**